enes-homework-week3-moreJS.js

```diff
@@ -0,0 +1,200 @@
+//Q:More JS
+
+//Q1
+
+function sum(a,b,c) {
+  return (a+b+c);
+};
+
+console.log(sum(1,2,3));
+
+//Q2
+
+function colorCar(str) {
+  console.log("a " + str + " car" );
+}
+
+const str1 = "red";
+
+console.log(colorCar(str1));
+
+
+//Q3
+const obj1 = {
+  name: "Jason",
+  surname: "Born",
+  age: 33,
+};
+
+function printOutPropertiesAndValues (obj) {
+  for (let key in obj) {
+    console.log(key);
+    console.log(obj[key]);
+  }
+}
+
+printOutPropertiesAndValues(obj1);// expected output: name surname age Jason Born 33
+
+Q4
+let vehicles = "";
+function codeDesigner(color, code) {//since one function is designed to do one job, I divided the question in two functions.
+  if (code === 1){
+    vehicles = "car";
+  } else if (code === 2){
+    vehicles = "motorcycle";
+  } else {
+    vehicles = "";
+    console.log("please select 1 for cars and 2 for motorcycles");
+  }
+}
+function vehicleType(color, code) {
+  codeDesigner(color, code);
+  console.log("a " + color + " " + vehicles);
+}
+
+vehicleType("blue", 1);
+
+  /* Another method: 
+  (However I could not find out how to write down  
+    "please select 1 for cars and 2 for motorcycles" 
+    when selected other than 1,2 with this method.)
+  function vehicleType(color, code) {
+    vehicles = (code ===1 ) ? "car" :
+              (code === 2 ) ? "motorcyle" : 
+              "";
+  console.log("a " + color + " " + vehicles);
+  }
+  */
+
+vehicleType("blue", 1);
```

Comment:
Hi Enes, couple things here:
1. I understand that you wanted to split out your code into two functions and I appreciate your thinking. However, I think the final solution is a bit complicated and could have been easier. Downfall of the approach is that you have to now create a sort-of global variable i.e `let vehicles = "";` which you then manipulate in `codeDesigner` function and it is then read inside the `vehicleType` function. I think I would have kept a single function for simplicity sake and inside the `if-elseif-else` had the console log statements thus avoiding the need of a new variable `vehicles`. 

2. Using conditional operators in this case isn't the best of ideas because of the reason which you yourself have mentioned in the comments. Maybe try using a `switch` block instead 🙂

Reply:
I could have created this variable inside the function. I divided the function in two because I wanted to use the function in the coming questions.